tests/test_directives.py

```diff
@@ -155,59 +180,122 @@ def test_set_undefined_directive(self, setUp, available_directives_list, product
-        directives[product_directive.name] = 100
+        directives['test'] = True
         directives._set_defined_directive(NP.name, 100)
         assert directives[NP.name] == 100
-        assert directives[product_directive.name] == 100
+        assert directives['test']
         del directives[NP.name]
         assert directives[NP.name] == NP.default
-        del directives[product_directive.name]
+        del directives['test']
         with pytest.raises(KeyError):
-            directives[product_directive.name]
+            directives['test']
 
-    def test_update_directive_without_aggregate(self, setUp, available_directives_list):
+    def test_update_directive_without_aggregate(self, setUp, available_directives_list, values):
         directives1 = setUp(available_directives_list=available_directives_list)
-        directives2 = deepcopy(directives1)
-        valid_values = {'np': 4, 'ngpu': 1, 'nranks': 0,
-                        'omp_num_threads': 10, 'executable': 'PathFinder',
-                        'walltime': 20., 'memory': 16, 'processor_fraction': 0.5}
-        for dirs in directives2:
-            directives2[dirs] = valid_values[dirs]
-        directives1.update(directives2)
+        valid_values_0 = values()
+        valid_values_1 = values(1)
         for dirs in directives1:
-            assert directives1[dirs] == directives2[dirs]
+            directives1[dirs] = valid_values_0[dirs]
+        expected_values = {'np': 100, 'ngpu': 10, 'nranks': 5,
+                           'omp_num_threads': 20, 'executable': 'Non Default Path',
+                           'walltime': 64.0, 'memory': 32, 'processor_fraction': 0.5}
+        directives1.update(valid_values_1)
+        for dirs in directives1:
+            assert directives1[dirs] == expected_values[dirs]
 
-    def test_update_directive_serial(self, setUp, available_directives_list):
+    def test_update_directive_serial(self, setUp, available_directives_list, values):
         directives1 = setUp(available_directives_list=available_directives_list)
         directives2 = deepcopy(directives1)
-        valid_values = {'np': 4, 'ngpu': 1, 'nranks': 0,
-                        'omp_num_threads': 10, 'executable': 'PathFinder',
-                        'walltime': 20., 'memory': 16, 'processor_fraction': 0.5}
-        expected_values = {'np': 10, 'ngpu': 1, 'nranks': 0,
-                           'omp_num_threads': 10, 'executable': sys.executable,
-                           'walltime': 32., 'memory': 16, 'processor_fraction': 1.}
+        valid_values_0 = values()
+        valid_values_1 = values(1)
+        expected_values = {'np': 100, 'ngpu': 10, 'nranks': 5,
+                           'omp_num_threads': 20, 'executable': 'Non Default Path',
+                           'walltime': 128.0, 'memory': 32, 'processor_fraction': 0.5}
+        for dirs in directives1:
+            directives1[dirs] = valid_values_0[dirs]
```

Comment:
```suggestion
        directives1.update(valid_values_0)
```
That functionality is already tested so you can just use it here.